frontend/src/metabase/collections/components/CollectionContent/CollectionContent.tsx

```diff
@@ -0,0 +1,111 @@
+import _ from "underscore";
+
+import { useSelector, useDispatch } from "metabase/lib/redux";
+import type {
+  CollectionId,
+  BookmarkId,
+  BookmarkType,
+} from "metabase-types/api";
+import type { UploadFileProps } from "metabase/redux/uploads";
+import { uploadFile as uploadFileAction } from "metabase/redux/uploads";
+
+import {
+  useCollectionQuery,
+  useCollectionListQuery,
+  useDatabaseListQuery,
+  useBookmarkListQuery,
+} from "metabase/common/hooks";
+
+import Bookmark from "metabase/entities/bookmarks";
+import Databases from "metabase/entities/databases";
+import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper";
+
+import { getUserIsAdmin } from "metabase/selectors/user";
+import { getSetting } from "metabase/selectors/settings";
+import { getIsNavbarOpen } from "metabase/selectors/app";
+
+import { CollectionContentView } from "./CollectionContentView";
+
+export function CollectionContent({
+  collectionId,
+}: {
+  collectionId: CollectionId;
+}) {
+  const { data: bookmarks, error: bookmarksError } = useBookmarkListQuery();
+  const { data: databases, error: databasesError } = useDatabaseListQuery();
+  const { data: collections, error: collectionsError } = useCollectionListQuery(
+    {
+      query: {
+        tree: true,
+        "exclude-other-user-collections": true,
+        "exclude-archived": true,
+      },
+    },
+  );
+  const { data: collection, error: collectionError } = useCollectionQuery({
+    id: collectionId,
+  });
+
+  const uploadDbId = useSelector(state =>
+    getSetting(state, "uploads-database-id"),
+  );
+  const uploadsEnabled = useSelector(state =>
+    getSetting(state, "uploads-enabled"),
+  );
+
+  const canUploadToDb = useSelector(
+    state =>
+      uploadDbId &&
+      Databases.selectors
+        .getObject(state, {
+          entityId: uploadDbId,
+        })
+        ?.canUpload(),
+  );
+
+  const isAdmin = useSelector(getUserIsAdmin);
+  const isNavbarOpen = useSelector(getIsNavbarOpen);
+
+  const dispatch = useDispatch();
+
+  const createBookmark = (id: BookmarkId, type: BookmarkType) =>
+    dispatch(Bookmark.actions.create({ id, type }));
+  const deleteBookmark = (id: BookmarkId, type: BookmarkType) =>
+    dispatch(Bookmark.actions.delete({ id, type }));
```

Comment:
!nit - don't feel like you have to change anything if you don't want to, but these hooks feel very technical.
something like this feels less cluttery imo:
```ts
const [uploadDbId, uploadsEnabled] = useSettings(["uploads-database-id", "uploads-enabled"]);
const { canUpload } = useDatabase(uploadDbId);
const { isAdmin } = useCurrentUser();
const { isOpen } = useNav():
const { create, delete } = useBookmarks();
```

similarly, the data fetching at the top of this render block is a pattern i keep seeing used. we fetch like 3-4 entities and then combine the values like `const error = bookmarksError || databasesError || collectionsError || collectionError;`. Feels like there's an opportunity to make this one hook that does this for us behind the scenes.

Reply:
I like the way you're thinking here, but that's a different PR